mail/api.py

```diff
@@ -151,6 +152,9 @@ def send_batch(cls, subject, body, recipients,  # pylint: disable=too-many-argum
                 'html': filter_recipient_variables(body),
                 'text': filter_recipient_variables(fallback_text),
                 'recipient-variables': json.dumps(chunk_dict),
+                'v:my-custom-data': {
```

Comment:
You need to use `json.dumps` here. Since `params` is not serialized to JSON each value here needs to be a string